File: src/my_cpp_pkg/src/number_counter.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"

class number_counter : public rclcpp::Node
{
    public:
        number_counter() : Node("number_counter")
        {
            RCLCPP_INFO(this->get_logger(), "Publisher node is initialised");
            subscriber_ = this->create_subscription<example_interfaces::msg::Int64>("number", 10, std::bind(&number_counter::callback_counter, this, std::placeholders::_1));
            publisher_ = this->create_publisher<example_interfaces::msg::Int64>("number_count", 10);
            
        }

    private:

        void callback_counter(const example_interfaces::msg::Int64 msg){
           publisher_->publish(msg);
        }

        rclcpp::Subscription<example_interfaces::msg::Int64>::SharedPtr subscriber_;
        rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr publisher_;
        rclcpp::TimerBase::SharedPtr timer_;
        int counter_ = 0;
};

int main(int argc, char **argv)
{

    rclcpp::init(argc, argv);
    auto node = std::make_shared<number_counter>();

    rclcpp::spin(node);

    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/number_counter.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import Int64

class number_counter(Node):
    def __init__(self):
        super().__init__("number_counter")

        self.subscriber_ = self.create_subscription(Int64, "number", self.callback_counter, 10)
        self.publisher_ = self.create_publisher(Int64, "number_count", 10)
        self.msg = Int64()
        self.get_logger().info("Counter node is initialised")

    def callback_counter(self, msg):
        self.publisher_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    
    node = number_counter()
    
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()